riscv/processor.cc

```diff
@@ -831,10 +814,9 @@ void processor_t::take_trap(trap_t& t, reg_t epc)
     s = set_field(s, MSTATUS_SPP, state.prv);
     s = set_field(s, MSTATUS_SIE, 0);
     state.sstatus->write(s);
-    set_privilege(PRV_S);
+    set_privilege(PRV_S, true);
```

Comment:
This looks correct, and also like a behavior change. Does it fix a bug, or am I misreading it?

Reply:
I don't think it's a behavior change. `vsdeleg` would be 0 unless `state.v` is already true, so we won't get here unless it's already true.